File: src/main/java/io/baris/petclinic/dropwizard/system/PetclinicConfiguration.java
package io.baris.petclinic.dropwizard.system;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.dropwizard.Configuration;
import io.dropwizard.client.HttpClientConfiguration;
import io.dropwizard.db.DataSourceFactory;
import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

/**
 * Vet service configuration
 */
@Data
public class PetclinicConfiguration extends Configuration {

    @NotEmpty
    String env;

    @Valid
    @NotNull
    private DataSourceFactory database = new DataSourceFactory();

    @NotNull
    private DatabaseConfig databaseConfig;

    @Valid
    @NotNull
    private HttpClientConfiguration httpClient = new HttpClientConfiguration();

    @JsonProperty("httpClient")
    public HttpClientConfiguration getHttpClientConfiguration() {
        return httpClient;
    }

    @JsonProperty("httpClient")
    public void setHttpClientConfiguration(HttpClientConfiguration httpClient) {
        this.httpClient = httpClient;
    }
}

File: src/main/kotlin/io/baris/petclinic/dropwizard/system/PetclinicConfiguration.kt
package io.baris.petclinic.dropwizard.system

import com.fasterxml.jackson.annotation.JsonCreator
import io.dropwizard.Configuration
import io.dropwizard.db.DataSourceFactory

/**
 * Pet Clinic service configuration
 */
data class PetclinicConfiguration @JsonCreator constructor(
    val env: String,
    val database: DataSourceFactory,
    val databaseConfig: DatabaseConfig
) : Configuration()
